assets/src/blocks/amp-story-text/edit.js

```diff
@@ -40,108 +43,190 @@ const applyFallbackStyles = withFallbackStyles( ( node, ownProps ) => {
 	};
 } );
 
-function TextBlock( props ) {
-	const {
-		attributes,
-		setAttributes,
-		className,
-		fontSize,
-		setFontSize,
-		backgroundColor,
-		textColor,
-		setBackgroundColor,
-		setTextColor,
-		fallbackTextColor,
-		fallbackBackgroundColor,
-	} = props;
+class TextBlockEdit extends Component {
+	componentDidUpdate() {
```

Comment:
@swissspidy I've added the amp-fit-text's replication logic here. Note that I didn't make any changes to `fontSizePicker`, so I renamed the PR back to WIP.

Testing this it feels like maybe it would be good if the user could still assign a custom font size, too. Perhaps we could even have just only two values for this: Auto (amp-fit-text logic) and Custom. Thoughts?

Also, let me know if you know of a better place for adding the logic, note that it needs to check the element's changing width and height to calculate the font size so the element should be rendered.

Reply:
> Testing this it feels like maybe it would be good if the user could still assign a custom font size, too.
> Note that I didn't make any changes to `FontSizePicker` (yet)

At the moment I see two reasonable options here:

* Add a new "Auto" option to font size dropdown \*
* Add a "Automatically fit text to container" `ToggleControl` (like in the post editor) that is enabled by default.  
  The font size picker would only be displayed when this toggle is disabled. Screenshot:

<img width="271" alt="Screenshot 2019-03-18 at 22 18 30" src="https://user-images.githubusercontent.com/841956/54564450-20906200-49cc-11e9-9679-95eaa34113a7.png">

\* Worth noting that the font options come from the `withFontSizes` HOC (and thus editor settings), not sure how easy it is to override these.

----

Besides this, I'd probably extract `calculateFontSize` in its own class method or even move it to `helper.js` as it is static.

Also, I am not sure if this should use `setFontSize()` or use a new attribute, e.g. `autoFontSize`. But that probably makes more sense when using option 2 from above (toggle). That way you can set a custom font size, turn on the toggle, turn it off, and get back your custom font size from before.

Reply:
> Add a "Automatically fit text to container" ToggleControl (like in the post editor) that is enabled by default.
The font size picker would only be displayed when this toggle is disabled.

Good idea, makes it consistent with other options too.

> Besides this, I'd probably extract calculateFontSize in its own class method or even move it to helper.js as it is static.

That's also a good point, especially since it would potentially be good to use it in case of other blocks at some point as well and not just with AMP Stories.

> That way you can set a custom font size, turn on the toggle, turn it off, and get back your custom font size from before.

Hmm. There are some cases where it might be unexpected to switch back to an old custom font size, however, there are more cases where it makes sense. 👍